lib/fluent/plugin_id.rb

```diff
@@ -59,5 +64,17 @@ def plugin_id
         "object:#{object_id.to_s(16)}"
       end
     end
+
+    def plugin_root_dir
+      return @_plugin_root_dir if @_plugin_root_dir
+      return nil unless system_config.root_dir
```

Comment:
Put `log.warn` is better?
Maybe, users hard to find the cause why `buffer_path`/`storage` is not configured automatically.